### ISSUE TITLE
{Row} not invoked on prop change?

### ISSUE THREAD

**Issue by damiangreen** [bvaughn/react-window#95]
i'm attempting to use this library to render a list of checkboxes, however when the props to my component change ,the {Row} function is not re-invoked, i.e. its only invoked once the first time.

Is this cached somehow?

Does this component work with changing data?

```  
   <List
            height="500"
            itemCount={this.items.length}
            itemSize={35}
            width={300}
          >
            {this.row}
          </List>
...
  private row = ({ index, style }: { index: number; style: any }) => {
    // tslint:disable-next-line:no-console
    console.log('row')
    return (
      <div style={style}>
        <Checkbox
          value={this.items[index].value}
          onChange={(e: any) => this.onChange(e, this.items[index].value)}
          checked={this.items[index].checked}
        >
          <span style={{ color: 'white' }}>
            {this.items[index].label} {this.items[index].checked}{' '}
          </span>
        </Checkbox>
      </div>
    )
  }
```

**Comment by bvaughn** [comment]
> Does this component work with changing data?

Yes, but in your above example— I don't see any props that would actually change (from List's perspective). Can you clarify what is actually changing?

**Comment by damiangreen** [comment]
ah sorry, so elements of  `this.items` have changing `checked` properties. I'e redux is storing a list of selected items and and i modify `this.items` on every change.

I expect not, but do i need to change the reference of the render prop each time my data changes?

**Comment by TrySound** [comment]
@damiangreen It's because `FixedSizeList` is a pure component. And nothing in props is changed in your example. Row is a component, not render function, so move it out of class and pass the list via `itemData` prop.

**Comment by damiangreen** [comment]
@TrySound That did the trick. Thanks.

I'm curious if there are any performance issues internally in the case i described with long lists and changing an individual element, e.g.checking checkboxes and having the List re-render each item . I'll experiment further but sure its's fine.
